src/store/DiscussionStore.ts

```diff
@@ -37,6 +37,14 @@ const useDiscussionStore = defineStore('discussion', () => {
     return data.value[subjectId.value]?.[page.value] ?? []
   })
 
+  /**
+   * Set the subjectId property.
+   */
+  function setId({ id }: { id: string | string[] }): string {
+    subjectId.value = Array.isArray(id) ? id[0] : id
```

Comment:
Pourquoi accepter un Array si de toute façon on ne prend que le premier élément et que le store ne fonctionne qu'avec un seul subjectId ?

Au passage (not this PR), pourquoi avoir introduit cette notion de subjectId au niveau racine du store ? Ca rend le store différent des autres, où on fait généralement `fetch({id})` et on stocke plusieurs résultats dans le store. C'est embêtant je trouve d'avoir plusieurs comportements différents en fonction des stores, et on part la fonctionnalité de "cache" si on affiche plusieurs fois les mêmes discussions lors d'une même navigation.

Reply:
Des bonnes questions.

> Pourquoi accepter un Array si de toute façon on ne prend que le premier élément et que le store ne fonctionne qu'avec un seul subjectId ?

Ça c'est juste lié au typage de `route.params.did`. L'alternative est soit ne pas faire attention au typage, soit faire du type casting du genre `id as string`. Ou autre, si tu as d'idées.

> Au passage (not this PR), pourquoi avoir introduit cette notion de subjectId au niveau racine du store ? Ca rend le store différent des autres, où on fait généralement fetch({id}) et on stocke plusieurs résultats dans le store.

Oui je me suis posé la même question, on peut je pense faire ce que tu dis (c'est plus simple.

> C'est embêtant je trouve d'avoir plusieurs comportements différents en fonction des stores

Ce serait bien de décider qu'est-ce qu'on attend des stores et de le documenter, je plussois.

> on part la fonctionnalité de "cache" si on affiche plusieurs fois les mêmes discussions lors d'une même navigation.

Je n'ai pas compris, si tu peux donner un exemple ce serait cool.

Reply:
Du coup, au moins pour cette PR, si ça te dit on supprime le `setId` et on remet le `fetch({id, page =1}` ?

Reply:
> Je n'ai pas compris, si tu peux donner un exemple ce serait cool.

Organisations > Clic sur une organisation > Affichage des jeux de données > Reviens sur la page organisation > Reclic sur la même organisation : les organisations et ses jeux de données ne sont récupérés qu'une fois depuis data.gouv.fr parce que les stores stockent les données.

Reply:
> Du coup, au moins pour cette PR, si ça te dit on supprime le setId et on remet le fetch({id, page =1} 

Vu que c'est pas introduit par cette PR j'hésite mais dans l'idée je plussoie 👍

Reply:
C'est une stacked PR donc je ne vois pas de problème de scope à ce qu'on fasse ce changement.

Reply:
Effectivement, j'avais pas encore démêlé la stack :-)